lib/SPIRV/SPIRVLowerBitCastToNonStandardType.cpp

```diff
@@ -160,23 +160,23 @@ class SPIRVLowerBitCastToNonStandardTypePass
         VectorType *SrcVecTy = getVectorType(BC->getSrcTy());
         if (SrcVecTy) {
           uint64_t NumElemsInSrcVec = SrcVecTy->getElementCount().getValue();
-          // SPV_INTEL_vector_compute allows 1-element vectors
+          // SPV_INTEL_vector_compute allows to use vectors with any number of
+          // components
           if (!isValidVectorSize(NumElemsInSrcVec) &&
-              !(Opts.isAllowedToUseExtension(
-                    ExtensionID::SPV_INTEL_vector_compute) &&
-                NumElemsInSrcVec == 1))
+              !Opts.isAllowedToUseExtension(
```

Comment:
In case if this extension is included we doesn't lowering any instructions with vector types.
Maybe to move this check just to the beginning of this method?

Reply:
I was just thinking that maybe this pass will lower not only vector instructions one day, so I left it here. Do you think it is possible?

Reply:
I think that we might rename this concrete method to use it for non standard vectors lowering only. And to write new functions for lowering other non supported patterns. It would make these changes atomic to each other.

Reply:
Ok, I've moved check to the beginning.